internal/bazel_integration_test/bazel_integration_test.bzl

```diff
@@ -0,0 +1,122 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Bazel integration testing
+"""
+
+# Helper function to convert a file to a path in the MANIFEST file
+def _file_to_manifest_path(ctx, f):
+    p = f.short_path
+    if p.startswith("../"):
+        return p[3:]
+    else:
+        return ctx.workspace_name + "/" + p
+
+def _bazel_integration_test(ctx):
+    # Serialize configuration file for test runner
+    config = ctx.actions.declare_file("%s.config.js" % ctx.attr.name)
+    ctx.actions.write(
+        output = config,
+        content = """// bazel_integration_test runner config generated by bazel_integration_test rule
+module.exports = {{
```

Comment:
maybe this could be a lot more readable if you create a struct of the data, then just use struct.to_json() to create the content?